spyder/widgets/helperwidgets.py

```diff
@@ -274,13 +275,33 @@ def update_status(self, value, value_set):
         self.update()
 
     def paintEvent(self, event):
-        """Qt Override.
-
-        Include a validation icon to the left of the line edit.
         """
-        super(IconLineEdit, self).paintEvent(event)
-        painter = QPainter(self)
+        Include a validation icon to the left of the line edit and elide text
+        if requested.
+        """
+        # Elide text if requested.
+        # See PR spyder-ide/spyder#20005 for context.
+        if self.elide_text and not self._focus_in:
+            # This code is taken for the most part from the
+            # AmountEdit.paintEvent method, part of the Electrum project.
```

Comment:
Should we add a link to the place on the Electrum project source code from where this was taken/is based?

Reply:
I mentioned that explicitly in the `NOTICE.txt` entry I created for Electrum:

https://github.com/spyder-ide/spyder/pull/20005/files#diff-b810ebe4ffda4293c79d292059b45b59de7e649e86314ffd5a1ee4eb4b1e08c8R187

But I could add it here too if you that's important.

Reply:
Yes please :+1:

Reply:
Ok, I added an extra line mentioning that for details people need to take a look at the Electrum entry in Preferences.